helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/InstancesAccessor.java

```diff
@@ -260,10 +271,44 @@ private Response batchGetStoppableInstances(String clusterId, JsonNode node, boo
         }
       }
 
+      if (node.get(InstancesProperties.stoppable_check_list.name()) != null) {
+        List<String> list = OBJECT_MAPPER.readValue(
+            node.get(InstancesProperties.stoppable_check_list.name()).toString(),
+            OBJECT_MAPPER.getTypeFactory().constructCollectionType(List.class, String.class));
+
+        if (ALL_STOPPABLE_CHECK_LIST.equals(list)) {
+          stoppableCheckList = HealthCheck.STOPPABLE_CHECK_LIST;
+        } else {
+          try {
+            stoppableCheckList =
+                list.stream().map(HealthCheck::valueOf).collect(Collectors.toList());
+          } catch (IllegalArgumentException e) {
+            String message =
+                "'stoppable_check_list' has invalid check names: " + list
+                    + ". Supported checks: " + HealthCheck.STOPPABLE_CHECK_LIST;
+            _logger.error(message, e);
+            return badRequest(message);
+          }
+        }
+      } else {
+        // By default, if stoppable_check_list is unset, all checks are performed to maintain
+        // backward compatibility with existing clients.
+        stoppableCheckList = HealthCheck.STOPPABLE_CHECK_LIST;
+      }
```

Comment:
This restructure the logic, it is not very clean. We can discuss offline.

Reply:
Logic is cleaned up as follow
- If the `STOPPABLE_CHECK_LIST` is not provided, Helix will perform **all Helix own stoppable checks** by default. And this will help with the backward compatibility since all existing users would expect Helix to perform all checks.
- If the `STOPPABLE_CHECK_LIST` is provided, 
  - If it's empty, Helix will skip all Helix own checks.
  - If it's not empty, Helix will only perform the given checks and carry out the checks in the order they are listed.

One more thing to note. In the current code, the HELIX:INSTANCE_NOT_EXIST will always be performed because I think it doesn't make sense to mark instances as stoppable if they don't even exist. 
Thanks for bringing this issue.